scripts/calculateRevenue/protocols/somm/dailySnapshots.ts

```diff
@@ -64,6 +68,27 @@ export function calculateWeightedAveragePrice(
     throw new Error('No snapshots provided')
   }
 
+  const sortedSnapshots = snapshots.sort(
+    (a, b) => new Date(a.timestamp).getTime() - new Date(b.timestamp).getTime(),
+  )
+  const firstSnapshotTime = new Date(sortedSnapshots[0].timestamp).getTime()
+  const lastSnapshotTime = new Date(
+    sortedSnapshots[sortedSnapshots.length - 1].timestamp,
+  ).getTime()
+
+  if (startTime < firstSnapshotTime) {
+    throw new Error('Start time is before the first snapshot')
+  }
+  if (endTime > lastSnapshotTime + TWENTY_FOUR_HOURS) {
+    throw new Error('End time is after the last snapshot')
+  }
+  if (
+    sortedSnapshots.length !==
+    (lastSnapshotTime - firstSnapshotTime) / TWENTY_FOUR_HOURS + 1
+  ) {
+    throw new Error('Missing snapshots')
+  }
```

Comment:
this check is duplicated for every time chunk right? since we pass the entire list of dailySnapshots.

Reply:
yea, not great. I'll move this to the api call